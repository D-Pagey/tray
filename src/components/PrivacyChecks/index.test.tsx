import React from 'react';
import { render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { PrivacyChecks, PrivacyChecksTypes } from '.';

const props: PrivacyChecksTypes = {
    onBack: () => null,
    onSubmit: () => null
};

describe('PrivacyChecks component', () => {
    it('should render', () => {
        const { container } = render(<PrivacyChecks {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should call onSubmit with both values when both checked', async () => {
        const onSubmit = jest.fn();
        const { getByTestId } = render(<PrivacyChecks {...props} onSubmit={onSubmit} />);

        userEvent.click(getByTestId('communicationCheckbox'));
        userEvent.click(getByTestId('updatesCheckbox'));
        userEvent.click(getByTestId('privacySubmit'));

        await waitFor(() =>
            expect(onSubmit).toHaveBeenCalledWith({ receiveUpdates: true, receiveCommunication: true })
        );
    });

    it.each`
        receiveUpdates
        ${true}
        ${false}
    `('should call onSubmit with correct values when receiveUpdates = $receiveUpdates', async ({ receiveUpdates }) => {
        const onSubmit = jest.fn();
        const { getByTestId } = render(<PrivacyChecks {...props} onSubmit={onSubmit} />);

        if (receiveUpdates) {
            userEvent.click(getByTestId('updatesCheckbox'));
        } else {
            userEvent.click(getByTestId('communicationCheckbox'));
        }

        userEvent.click(getByTestId('privacySubmit'));

        await waitFor(() =>
            expect(onSubmit).toHaveBeenCalledWith({
                receiveUpdates,
                receiveCommunication: !receiveUpdates
            })
        );
    });
});
