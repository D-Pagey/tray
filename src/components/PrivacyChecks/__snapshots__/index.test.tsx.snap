// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivacyChecks component should render 1`] = `
<form
  class="sc-AxirZ jfPcbx"
>
  <div
    class="sc-AxiKw fKWGNZ"
  >
    <input
      data-testid="updatesCheckbox"
      type="checkbox"
    />
    <p
      class="sc-AxhCb bYDHFA"
    >
      Receive updates about Tray.io product by email
    </p>
  </div>
  <div
    class="sc-AxiKw fKWGNZ"
  >
    <input
      data-testid="communicationCheckbox"
      type="checkbox"
    />
    <p
      class="sc-AxhCb bYDHFA"
    >
      Receive communication by email for other products created by the Tray.io team
    </p>
  </div>
  <button
    class="sc-AxjAm jyhTFg"
    data-testid="privacySubmit"
    type="submit"
  >
    Next
  </button>
</form>
`;
